src/test/java/test/CheckSuitesInitializationTest.java

```diff
@@ -33,8 +34,8 @@ public void check() {
     TestListenerAdapter tla = new TestListenerAdapter();
     TestNG tng = create();
     String testngXmlPath = getPathToResource("checksuitesinitialization/parent-suite.xml");
-    tng.setTestSuites(Arrays.asList(testngXmlPath));
-    tng.addListener(tla);
+    tng.setTestSuites(Collections.singletonList(testngXmlPath));
+    tng.addListener((ITestNGListener) tla);
```

Comment:
Casting should be useless now because you should have removed others addListener methods

Reply:
In fact, I didn't find the removal of deprecated addListener. Could you remove or hide them?

Reply:
I didn't remove `addListener()` because there's a warning saying I need to check with someone at IntelliJ before I remove it off. Please see [here](https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/TestNG.java#L610-L618)

Reply:
@akozlova We'd like to remove deprecated code for the next TestNG major release. Is the `TestNG#addListener(Object)` ok for you?

Reply:
Yes, we use explicitly this method from the overloaded family. Thanks!

Reply:
@akozlova Sorry, I forgot a word...

Is the TestNG#addListener(Object) removal ok for you? At least with another visibility instead of public.

Reply:
Protected would work but if it would be removed completely, I would have to add tons of reflection... We are stuck with 6.8 as don't want to enforce users to work with java 8. It's possible to split support into parts according to the user's version but that's some work I'd like to see benefits of. Sorry

Reply:
protected is enough, thanks for the feedback :)